src/network/networkd-link.c

```diff
@@ -2337,6 +2354,7 @@ static int link_initialized_and_synced(Link *link) {
                 }
 
                 link->network = network_ref(network);
+                link_update_operstate(link, false);
```

Comment:
I do not think this is necessary.

Reply:
I put this here for managed links which have no carrier at boot. `link_update_operstate()` needs to get called at least once after the `link->network` pointer has been set, so that the online state can transition from `_LINK_ONLINE_STATE_INVALID` to `LINK_ONLINE_STATE_OFFLINE`.

Put another way, we want managed links which are offline to be marked as such. Without this line, such links which never have carrier may be marked as "n/a" (or "unknown", etc.).

I'm not sure I see an alternative, except to have the default online state of a link be `LINK_ONLINE_STATE_OFFLINE`. But I am trying to make networkd as transparent as possible when it advertises the online state. Hence I prefer to say "networkd has no idea", than "networkd thinks this is offline".